tests/api/enqueue.cpp

```diff
@@ -88,3 +88,52 @@ TEST_F(WithCommandQueue, KernelNoArguments) {
     clReleaseEvent(event);
     ASSERT_EQ(status, CL_COMPLETE);
 }
+
+TEST_F(WithCommandQueue, WorkDim) {
```

Comment:
This is covered by a conformance test but it also covers a bunch of other functions as well. Good to have as a unit test/(base for a )benchmark.